### ISSUE TITLE
Apply entity builder on setter.

### ISSUE THREAD

**Issue by guisouza** [scup/speck#30]
Guys I really hope I am wrong, but i think we are no applying entity builders on setter, but just on entity construction ..  

Is this the expected behaviour ? or it is a bug ? 

@albertossilva could you please check, i have a fix for it, like this : 
```javascript
    set: function (value){
      if (instance.schema[field].type || instance.schema[field].builder) {
        value = instance.applyEntityConstructor(instance.schema[field], value);
      }
      if(instance.data[field] !== value) {
        instance.data[field] = value;
        return instance._validate();
      }
    },
```

**Comment by albertossilva** [comment]
@guisouza Yes, we are not applying it is nice improvement... I would also check if value is already instanceOf `instance.schema[field].type`.
And.... let's avoid the value override at the second line please, ahahaa, 😛

**Comment by guisouza** [comment]
good point ..  First of all, check if its already instantiated .. 

Now thinking ..  should we pop an error in cases which the constructor of a entity returned by the builder differs from the declared Type ?  
Or when we have a custom builder, the Type is just a documentation, IMHO, No! after all we are passing the Type to every builder .. 

@albertossilva its getting complex .. #helpwanted
